### PR TITLE
added http const. fixes #1466

### LINKED ISSUE #1466: HTTP Status Codes (302, 501) names are incorrect
In reading the docs for this module, I noticed that the HTTP status codes have minor naming issues.

* ngx.HTTP_MOVED_TEMPORARILY (302) should be [ngx.HTTP_FOUND](https://en.wikipedia.org/wiki/HTTP_302)
* ngx.HTTP_METHOD_NOT_IMPLEMENTED (501) should be [ngx.HTTP_NOT_IMPLEMENTED](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_errors)

### PR DESCRIPTION
**Pull request by karai17**
* Add HTTP_NOT_IMPLEMENTED (501)

HTTP_METHOD_NOT_IMPLEMENTED kept for compatibility.

I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

**Comment by agentzh** [comment]
@karai17 Travis CI jobs are all red for this PR: https://travis-ci.org/openresty/lua-nginx-module/builds/488131841?utm_source=github_status&utm_medium=notification

File: src/ngx_http_lua_consts.c [hunk 1]
```diff
@@ -178,6 +178,10 @@ ngx_http_lua_inject_http_consts(lua_State *L)
     lua_pushinteger(L, NGX_HTTP_INTERNAL_SERVER_ERROR);
     lua_setfield(L, -2, "HTTP_INTERNAL_SERVER_ERROR");
 
+    lua_pushinteger(L, NGX_HTTP_NOT_IMPLEMENTED);
+    lua_setfield(L, -2, "HTTP_NOT_IMPLEMENTED");
```
Comment (agentzh) [inline src/ngx_http_lua_consts.c]: Need to adjust the initial size of the Lua table since we're adding new entries to it.

**Comment by agentzh** [comment]
@thibaultcha Please help me follow up this PR :) Thanks!

**Review by zhuizhuhaomeng — APPROVED** [review]

**Comment by zhuizhuhaomeng** [comment]
merged with the following patch.
```
diff --git a/src/ngx_http_lua_util.c b/src/ngx_http_lua_util.c
index bae59daf..232a1c13 100644
--- a/src/ngx_http_lua_util.c
+++ b/src/ngx_http_lua_util.c
@@ -827,7 +827,7 @@ static void
 ngx_http_lua_inject_ngx_api(lua_State *L, ngx_http_lua_main_conf_t *lmcf,
     ngx_log_t *log)
 {
-    lua_createtable(L, 0 /* narr */, 113 /* nrec */);    /* ngx.* */
+    lua_createtable(L, 0 /* narr */, 115 /* nrec */);    /* ngx.* */
 
     lua_pushcfunction(L, ngx_http_lua_get_raw_phase_context);
     lua_setfield(L, -2, "_phase_ctx");
diff --git a/t/062-count.t b/t/062-count.t
index 598441c5..f5ba12c9 100644
--- a/t/062-count.t
+++ b/t/062-count.t
@@ -34,7 +34,7 @@ __DATA__
 --- request
 GET /test
 --- response_body
-ngx: 114
+ngx: 115
 --- no_error_log
 [error]
 
@@ -55,7 +55,7 @@ ngx: 114
 --- request
 GET /test
 --- response_body
-114
+115
 --- no_error_log
 [error]
 
@@ -83,7 +83,7 @@ GET /test
 --- request
 GET /test
 --- response_body
-n = 114
+n = 115
 --- no_error_log
 [error]
 
@@ -305,7 +305,7 @@ GET /t
 --- response_body_like: 404 Not Found
 --- error_code: 404
 --- error_log
-ngx. entry count: 114
+ngx. entry count: 115
```

**Pull request closed without merging**